evalml/tests/component_tests/decomposer_tests/test_decomposer.py

```diff
@@ -17,8 +29,455 @@ def test_set_time_index():
+        "mid period": 3 * period + 4,
+    }[test_first_index]
+
+    # Data spanning 2021-01-01 to 2021-02-09
+    X, _, y = ts_data()
+
+    # Change the date time index to start at the same time but have different frequency
+    y.set_axis(
+        pd.date_range(start="2021-01-01", periods=len(y), freq=frequency),
+        inplace=True,
+    )
+
+    decomposer = decomposer_child_class(degree=2)
+
+    # Synthesize a one-week long cyclic signal
+    single_period_seasonal_signal = np.sin(y[0:period] * 2 * np.pi / len(y[0:period]))
+    full_seasonal_signal = np.sin(y * 2 * np.pi / len(y[0:period]))
+
+    # Split the target data.  Since the period of this data is 7 days, we'll test
+    # when the cycle begins, an index before it begins, an index after it begins
+    # and in the middle of a cycle
+    y_test = y[test_first_index:]
+
+    # Set the decomposer's trend attribute since the function uses it to select the
+    # proper integer/datetime index.  The actual value doesn't matter, just that
+    # something with an index exists there.
+    decomposer.trend = full_seasonal_signal
+
+    projected_seasonality = decomposer._project_seasonal(
+        y_test,
+        single_period_seasonal_signal,
+        period,
+        frequency,
+    )
+
+    # Make sure that the function extracted the correct portion of the repeating, full seasonal signal
+    assert np.allclose(
+        full_seasonal_signal[projected_seasonality.index],
+        projected_seasonality,
+    )
+
+
+@pytest.mark.parametrize(
+    "decomposer_child_class",
+    decomposer_list,
+)
+def test_decomposer_get_trend_dataframe_raises_errors(
+    decomposer_child_class,
+    ts_data,
+):
+    X, _, y = ts_data()
+    dec = decomposer_child_class()
+    dec.fit_transform(X, y)
+
+    with pytest.raises(
+        TypeError,
+        match="Provided X should have datetimes in the index.",
+    ):
+        X_int_index = X.reset_index()
+        dec.get_trend_dataframe(X_int_index, y)
+
+    with pytest.raises(
+        ValueError,
+        match="Provided DatetimeIndex of X should have an inferred frequency.",
+    ):
+        X.index.freq = None
+        dec.get_trend_dataframe(X, y)
+
+
+@pytest.mark.parametrize(
+    "decomposer_child_class",
+    decomposer_list,
+)
+@pytest.mark.parametrize("period", [7, 30, 365])
+def test_decomposer_set_period(decomposer_child_class, period, generate_seasonal_data):
+    X, y = generate_seasonal_data(real_or_synthetic="synthetic")(period)
+    dec = decomposer_child_class()
+
+    if isinstance(dec, STLDecomposer):
+        assert dec.seasonal_period == 7
+        assert dec.parameters["seasonal_period"] == 7
+    elif isinstance(dec, PolynomialDecomposer):
+        assert dec.seasonal_period == -1
+        assert dec.parameters["seasonal_period"] == -1
+
+    dec.set_seasonal_period(X, y)
+
+    assert 0.95 * period <= dec.seasonal_period <= 1.05 * period
+    assert dec.parameters["seasonal_period"]
```

Comment:
Is this purposely just checking that the parameter exists? Should we be checking its value?

Reply:
OK, yes, this one is a bug.  For some reason, parameters dict value is not being set.  I tried fixing it, but for some reason cannot get it to set. I filed [this](https://github.com/alteryx/evalml/issues/3771) to look at it later given that we don't even use `set_seasonal_period`.